src/View/mainpage/ManipulateROIWindow.py

```diff
@@ -0,0 +1,529 @@
+
+            # Execute the selected operation
+            new_geometry = ROI.manipulate_rois(roi_1_geometry, roi_2_geometry,
+                                               uid_list,
+                                               selected_operation.upper())
+            self.new_ROI_contours = ROI.geometry_to_roi(new_geometry)
+
+            self.draw_roi()
+            return
+
+        QMessageBox.about(self.manipulate_roi_window_instance,
+                          "Not Enough Data",
+                          "Not all values are specified or correct.")
+
+    def onSaveClicked(self):
+        """ Save the new ROI """
+        # Get the name of the new ROI
+        new_roi_name = self.new_roi_name_line_edit.text()
+
+        if self.new_ROI_contours is None:
+            QMessageBox.about(self.manipulate_roi_window_instance,
+                              "Empty ROI", "No new ROI has been drawn")
+            return
+
+        # Get the required info to create the new ROI
+        slider_id = self.dicom_preview.slider.value()
+        dataset = self.patient_dict_container.dataset[slider_id]
+        location = self.patient_dict_container.filepaths[slider_id]
+        ds = pydicom.dcmread(location)
+        pixlut = self.patient_dict_container.get("pixluts")
+        pixlut = pixlut[dataset.SOPInstanceUID]
+        z_coord = dataset.SliceLocation
+
+        # Get the new contour sequence
+        if dataset.SOPInstanceUID not in self.new_ROI_contours:
+            # TODO: delete after merged with ROI create multiple
+            QMessageBox.about(self.manipulate_roi_window_instance,
+                              "Empty slice",
+                              "The current slice doesn't contain new ROI"
+                              "\nPlease move to other slices")
+            return
+        contour_sequence = self.new_ROI_contours[dataset.SOPInstanceUID]
+
+        # Transform the contour sequence to a 1D array
+        # Convert the pixel points to RCS points, then append z coordinate
+        single_array = []
+        for contour_data in contour_sequence:
+            for point in contour_data:
+                rcs_pixels = ROI.pixel_to_rcs(pixlut,
+                                              round(point[0]),
+                                              round(point[1]))
+                single_array.append(rcs_pixels[0])
+                single_array.append(rcs_pixels[1])
+                single_array.append(z_coord)
+
+        # Create a popup window that modifies the RTSTRUCT and tells the user
+        # that processing is happening.
+        progress_window = SaveROIProgressWindow(self,
+                                                QtCore.Qt.WindowTitleHint)
+        progress_window.signal_roi_saved.connect(self.roi_saved)
+        progress_window.start_saving(self.dataset_rtss, new_roi_name,
+                                     single_array, ds)
+        progress_window.show()
```

Comment:
this part looks similar to the save_roi function in draw roi window, so please either refactor this one or refactor the one in draw roi window, or create a function and call it here and in draw roi window.

Reply:
yes, I sure will refactor this issue when I merge this branch with master. It's already in my plan. Thanks for your comments.